plugin-sftp/src/main/java/ca/on/oicr/gsi/shesmu/sftp/SftpServer.java

```diff
@@ -121,6 +132,51 @@ public void configuration(SectionRenderer renderer) throws XMLStreamException {
     renderer.line("Active", connection.get().isPresent() ? "Yes" : "No");
   }
 
+  synchronized Pair<ActionState, Boolean> makeSymlink(
+      String link, String target, boolean fileInTheWay) {
+    final Optional<SFTPClient> client = connection.get().map(Pair::second);
+    if (!client.isPresent()) {
+      return new Pair<>(ActionState.UNKNOWN, fileInTheWay);
+    }
+    try {
+      final SFTPClient sftp = client.get();
+      // Because this library thinks that a file not existing is an error state worthy of exception,
+      // it throws whenever stat or lstat is called. There's a wrapper for stat that catches the
+      // exception and returns null, but there's no equivalent for lstat, so we reproduce that catch
+      // logic here.
+      try {
+        final FileAttributes attributes = sftp.lstat(link);
+        // File exists and it is a symlink
+        if (attributes.getType() == FileMode.Type.SYMLINK) {
+          // It's what we want; done
+          if (sftp.readlink(link).equals(target)) {
+            return new Pair<>(ActionState.SUCCEEDED, false);
+          } else {
+            // It's pointing to something else; recreate it
+            sftp.rm(link);
```

Comment:
This behaviour is different than ln -s and might be worth re-thinking. There's a possibility for it to completely trash the target symlink in an automated way.

Reply:
The other potential situation where this happens is when you have two files named the same that are headed for the same destination, e.g. two bam files from the same sample by different aligners.

Reply:
Yeah, I was making it more like `ln -sf`. I'll change this to just be an error.